### PR TITLE
docs(button, checkbox, input, label, radio-button-group): add usage examples, update demo html for A11y audit

### PR DESCRIPTION
**Pull request by caripizza**
**Related Issue:** #2149

## Summary
This pr adds usage examples for calcite-checkbox and calcite-radio-button-group, updates usage examples for calcite-button, calcite-input, and calcite-label, and updates the corresponding demo html pages for the initial LA audit of components (per the scope kickoff meeting we had last Thursday).

Also adds robots.txt no-index for the www folder output.

After this is approved and merged to master, I need to
1. [ ] ask Paul to redeploy `www` to Netlify
2. [ ] let LA know the updated pages are ready for them to begin testing

cc @paulcpederson

### CODE REVIEW

File: src/components/calcite-button/usage/as-anchor.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+```html
+<calcite-button href="#content">Internal anchor</calcite-button>
+<calcite-button href="https://www.esri.com" icon-start="launch" rel="noopener noreferrer" target="_blank">
```
Comment (driskull) [inline src/components/calcite-button/usage/as-anchor.md:3]: The more I look at this code, the more I think we should just have a slotted anchor to achieve this. Do we really want to add all the anchor props to the calcite button in order to support them or should we just allow a user to slot a link?

File: stencil.config.ts [hunk 2]
```diff
@@ -75,7 +75,10 @@ export const create: () => Config = () => ({
       type: "www",
       baseUrl: "https://stenciljs.com/",
       prerenderConfig: "./prerender.config.ts",
-      copy: [{ src: "demos", dest: "demos" }],
+      copy: [
+        { src: "demos", dest: "demos" },
+        { src: "robots.txt", dest: "robots.txt" }
```
Comment (driskull) [inline stencil.config.ts:80]: 🤖

**Review by driskull — APPROVED** [review]
👍 Nice @caripizza 💯

File: src/components/calcite-button/usage/as-anchor.md (hunk 1 above)
Comment (driskull) [reply src/components/calcite-button/usage/as-anchor.md:3]: The problem right now is that `rel="noopener noreferrer" target="_blank"` are not valid documented props for calcite-button

Comment (caripizza) [reply src/components/calcite-button/usage/as-anchor.md:3]: Gotcha @driskull I'll remove this usage suggestion (just realized this use case is already covered by internals.md)

File: src/components/calcite-radio-button-group/usage/basic.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+```html
+<calcite-radio-button-group name="group-basic">
+  <calcite-radio-button value="maps" checked>Maps</calcite-radio-button>
+  <calcite-radio-button value="layers">Layers</calcite-radio-button>
+  <calcite-radio-button value="data">Data</calcite-radio-button>
+  <calcite-radio-button value="geography">Geography</calcite-radio-button>
```
Comment (eriklharper) [inline src/components/calcite-radio-button-group/usage/basic.md:6]: `calcite-radio-button` no longer supports slotting in label content, so you should add wrapping `calcite-label`s with `layout="inline"` on them here.

File: src/demos/calcite-radio-button-group.html [hunk 4]
```diff
@@ -48,6 +52,58 @@
...
+
+        <section>
+          <!-- src/components/calcite-radio-button-group/usage/.md -->
+          <div>
+            <h3>Basic</h3>
+            <calcite-radio-button-group name="group-basic">
+              <calcite-radio-button value="maps" checked>Maps</calcite-radio-button>
+              <calcite-radio-button value="layers">Layers</calcite-radio-button>
+              <calcite-radio-button value="data">Data</calcite-radio-button>
+              <calcite-radio-button value="geography">Geography</calcite-radio-button>
```
Comment (eriklharper) [inline src/demos/calcite-radio-button-group.html:64]: ☝️ same here, these should be updated to have wrapping `calcite-label`s.

**Review by eriklharper — CHANGES_REQUESTED** [review]

File: src/components/calcite-radio-button-group/usage/basic.md (hunk 3 above)
Comment (caripizza) [reply src/components/calcite-radio-button-group/usage/basic.md]: done!

File: src/demos/calcite-radio-button-group.html (hunk 4 above)
Comment (caripizza) [reply src/demos/calcite-radio-button-group.html]: (also done)

**Pull request merged**